woocommerce-delivery-notes/includes/class-wcdn-theme.php

```diff
@@ -116,27 +116,44 @@ public function create_print_button_order_page( $order_id ) {
 		 * @param boolean $plain_text Whether only to send plain text email or not.
 		 */
 		public function add_email_print_url( $order, $sent_to_admin = true, $plain_text = false ) {
-			if ( 'yes' === get_option( 'wcdn_email_print_link' ) ) {
-				$wdn_order_billing_id = ( version_compare( get_option( 'woocommerce_version' ), '3.0.0', '>=' ) ) ? $order->get_billing_email() : $order->billing_email;
-				if ( $wdn_order_billing_id && ! $sent_to_admin ) {
-					$wdn_order_id = ( version_compare( get_option( 'woocommerce_version' ), '3.0.0', '>=' ) ) ? $order->get_id() : $order->id;
-
-					$url = wcdn_get_print_link( $wdn_order_id, $this->get_template_type( $order ), $wdn_order_billing_id, true );
+			$wdn_order_billing_id = ( version_compare( get_option( 'woocommerce_version' ), '3.0.0', '>=' ) ) ? $order->get_billing_email() : $order->billing_email;
 
-					if ( $plain_text ) :
-						echo esc_html_e( 'Print your order', 'woocommerce-delivery-notes' ) . "\n\n";
+			$wdn_order_id = ( version_compare( get_option( 'woocommerce_version' ), '3.0.0', '>=' ) ) ? $order->get_id() : $order->id;
 
-						echo esc_url( $url ) . "\n";
+			$url = wcdn_get_print_link( $wdn_order_id, $this->get_template_type( $order ), $wdn_order_billing_id, true );
 
-						echo "\n****************************************************\n\n";
-					else :
-						?>
-						<p><strong><?php esc_attr_e( 'Print:', 'woocommerce-delivery-notes' ); ?></strong> <a href="<?php echo esc_url_raw( $url ); ?>"><?php esc_attr_e( 'Open print view in browser', 'woocommerce-delivery-notes' ); ?></a></p>
-					<?php endif;
+			if ( 'yes' === get_option( 'wcdn_email_print_link' ) && 'yes' === get_option( 'wcdn_admin_email_print_link' ) ) {
+				$this->print_link_in_email( $plain_text, $url );
+			} elseif ( 'yes' === get_option( 'wcdn_email_print_link' ) ) {
+				if ( ( $wdn_order_billing_id && ! $sent_to_admin ) ) {
+					$this->print_link_in_email( $plain_text, $url );
+				}
+			} elseif ( 'yes' === get_option( 'wcdn_admin_email_print_link' ) ) {
+				if ( $sent_to_admin ) {
+					$this->print_link_in_email( $plain_text, $url );
```

Comment:
In above code block, remove the elseif blocks and the first if block. Simply use 2 different if blocks for customer & admin emails.